e2e/portalicious/tests/RegistrationPage/InitiateActions.spec.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  getRegistrationIdByReferenceId,
+  seedIncludedRegistrations,
+} from '@121-service/test/helpers/registration.helper';
+import {
+  getAccessToken,
+  resetDB,
+} from '@121-service/test/helpers/utility.helper';
+import {
+  programIdPV,
+  registrationPV5,
+} from '@121-service/test/registrations/pagination/pagination-data';
+
+import LoginPage from '@121-e2e/portalicious/pages/LoginPage';
+import RegistrationActivityLogPage from '@121-e2e/portalicious/pages/RegistrationActivityLogPage';
+
+const projectId = 2;
+let registrationId: number;
+
+test.beforeEach(async ({ page }) => {
+  await resetDB(SeedScript.nlrcMultiple);
+
+  const accessToken = await getAccessToken();
+  await seedIncludedRegistrations([registrationPV5], programIdPV, accessToken);
+  registrationId = await getRegistrationIdByReferenceId({
+    programId: programIdPV,
+    referenceId: registrationPV5.referenceId,
+    accessToken,
+  });
+
+  // Login
+  const loginPage = new LoginPage(page);
+  await page.goto(`/`);
+  await loginPage.login(
+    process.env.USERCONFIG_121_SERVICE_EMAIL_ADMIN,
+    process.env.USERCONFIG_121_SERVICE_PASSWORD_ADMIN,
+  );
+});
+
+test('User can initiate registration status changes from registration page', async ({
+  page,
+}) => {
+  const activityLogPage = new RegistrationActivityLogPage(page);
+
+  const statusChangeActions = ['Validate', 'Include', 'Decline', 'Delete'];
+
+  for (const actionName of statusChangeActions) {
+    await test.step('Navigate to registration', async () => {
+      await activityLogPage.goto(
+        `/project/${projectId}/registrations/${registrationId}`, // TODO should we not hardcode the url here?
+      );
+    });
+
+    await test.step(`Initiate action: ${actionName}`, async () => {
+      await activityLogPage.inititateAction(actionName);
+
+      const dialog = page.getByRole('dialog');
+      await expect(dialog).toBeVisible();
+
+      const title = dialog.getByText(`${actionName} registration(s)`);
+      await expect(title).toBeVisible();
+
+      // Check if the action is only for 1 registration
+      const paragraph = dialog.getByText(/You're about to/);
+      await expect(paragraph).toContainText(
+        `You're about to ${actionName.toLowerCase()} 1 registrations.`,
+      );
+    });
+  }
+});
+
+test('User can open add note sidebar from action menu', async ({ page }) => {
+  const activityLogPage = new RegistrationActivityLogPage(page);
+
+  const actionName = 'Add note';
+
+  await test.step('Navigate to registration', async () => {
+    await activityLogPage.goto(
+      `/project/${projectId}/registrations/${registrationId}`,
+    );
+  });
+
+  await test.step(`Initiate action: ${actionName}`, async () => {
+    await activityLogPage.inititateAction(actionName);
+    await page.waitForLoadState('domcontentloaded');
+    await page.waitForLoadState('networkidle');
+    const sidebarDescription = page.getByTestId('add-note-sidebar-subtitle');
```

Comment:
Below you did something similar with

```ts
const paragraph = dialog.getByText(/You're about to/);
```

which I think is better because it means we don't have to add a testid, and define a new element in the page.